### PR TITLE
Relops1077: Add Azure Shared Image Gallery support 

### PR DESCRIPTION
**Pull request by markcor**
WIP

### CODE REVIEW

File: .worker-images.yml.swp
Comment (jwmoss) [inline .worker-images.yml.swp:1]: Is this required?

File: .worker-pools.yml.swp
Comment (jwmoss) [inline .worker-pools.yml.swp:1]: Is this required?

**Comment by markcor** [comment]
There are four items I am trying to address here:

1. Creation of Arm64 Alpha builder pool
2. Removal of the unused/no longer needed code in worker_pools.py.
3. Add support to use images from the shared image gallery. This includes the introduction of image versions.
4. Add support to include the SBOM/manifest URL in the GW config.
  
@ahal Any suggestions on the following:

How to handle the bare except that pre-commit is failing.

I want to add the SBOM/manifest URL to the GW config for Windows images. This would be under the `workerTypeMetaData` in the GW config:

```
"workerTypeMetadata": {  
  "ImageManifest": "manifest URL"
  },
```
It should probably go around here in the `get_azure_provider_config` function:

```
                "workerConfig": merge(
                    worker_config,
                ),
```
However, I am not sure on the how. Also, note that part of this will be a temporary measure until we have all Windows images hosted in a Shared Image Gallery. Then some of the conditional language can go away, and we will likely construct manifest URL instead of hardcoding it.

File: src/ciadmin/generate/worker_pools.py [hunk 1]
```diff
@@ -201,109 +201,77 @@ def get_azure_provider_config(
...
+
+    launch_configs = []
+    for location in locations:
+        for vmSize in vmSizes:
+            loc = location.replace("-", "")
+            DeploymentId = image.image_id(provider_id, "deployment_id")
+            try:
+                manifest = image.image_id(provider_id, "manifest")
+            except:
+                pass
```
Comment (ahal) [inline src/ciadmin/generate/worker_pools.py:213]: This `pass` will cause problems later because `manifest` (and `version`) will actually be undefined. So even referencing them will be a runtime error. Instead try doing `manifest = None`.

File: src/ciadmin/generate/worker_pools.py [hunk 2]
```diff
@@ -201,109 +201,77 @@ def get_azure_provider_config(
...
+            subscription_id = f"/subscriptions/{subscription_id}"
+            resource_suffix = f"{location}-{purpose}"
+            rgroup = f"rg-{resource_suffix}"
+            vnet = f"vn-{resource_suffix}"
+            snet = f"sn-{resource_suffix}"
+            subnetId = (
+                f"{subscription_id}/resourceGroups/{rgroup}/providers/"
+                f"Microsoft.Network/virtualNetworks/{vnet}/subnets/{snet}"
+            )
+            if "version" in locals() and version:
```
Comment (ahal) [inline src/ciadmin/generate/worker_pools.py:231]: Oh I see, you're checking that it's defined here. Probably still better to just set it to `None` in the `except` clause and then change this to `if version:`.

File: src/ciadmin/generate/worker_pools.py [hunk 3]
```diff
@@ -201,109 +201,77 @@ def get_azure_provider_config(
             "wstServerURL", azure_config["wst_server_url"]
         )
     tags = config.get("tags", {})
-    # Temporary conditional language while we move pools over to the newer
-    # method that is in the else clause
-    # Refrence https://mozilla-hub.atlassian.net/browse/RELOPS-212
-    if config.get("old-style-worker-config"):
```
Comment (ahal) [inline src/ciadmin/generate/worker_pools.py:207]: Thanks for deleting this! Though it would be nice to remove this in a separate PR (or at least a separate commit) to make the diff easier to read.

**Review by ahal — CHANGES_REQUESTED** [review]

File: src/ciadmin/generate/worker_pools.py (hunk 3 above)
Comment (markcor) [reply src/ciadmin/generate/worker_pools.py:207]: Good point. I am going to separate this into different PRs.
